### DISCUSSION TITLE
Past commit histories showing different account.

### DISCUSSION THREAD

**Discussion by serenagrl** [community/community#72601] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Good day,

I am currently experiencing an issue and I am not sure whether it is a bug and whether it can be rectified.

I have an account (let's call it **Account-1**) with 4 repositories over the past few years and I have been actively working on 1 repository for more than a year with another contributor. About a month ago, I created a NEW account (**Account-2**) for other learning and testing purposes for my job but that account was not used in any repositories owned by Account-1. When working on my repo, I used Account-1 to sign in.

While checking the **Commit History** and **Insights** of my repositories today, I have found out that some of my past commit histories dated to a year back (2022) was replaced with Account-2's name. I checked my other repository and found that one of them also exhibited the same problem with histories being replaced to Account-2's name as far as year 2021.

I am not sure how a new account that is created only a month ago and not being invited or part of the contributors can replaced the past histories randomly that spanned 2 years back in my repos. 

Some information to aid the forensics:

1. Both Account-1 and Account-2 are registered using different emails. 
2. `git config user.name` and `git config user.email` both points to Account-1
3. `git config --global user.name` and `git config --global user.email` both points to Account-1
4. Account-2 is not part of any project.
5. Histories being replaced are not continuous, it looks random.
6. I am using Visual Studio code for the current repo which has its histories replaced as far as 2022.
7. I used Visual Studio Enterprise edition for the dormant repo which got its histories replaced as far as 2021. I have not updated this repo for more than 2 years but the histories got replaced with Account-2.

I hope someone can help me figure this out.

Thank You.

Best Regards,
Serena

**Comment by hshdfs** [comment]
It seems like you're encountering an unusual issue with your Git commit history and account usage. The problem appears to involve some of your past commit histories being replaced with Account-2's name, even though that account is relatively new and unrelated to the repositories in question. You've provided some useful information to aid in troubleshooting the issue, including:

Both accounts (Account-1 and Account-2) using different email addresses.
Confirmation that your Git configurations (user.name and user.email) are correctly set to Account-1.
Account-2 not being part of any project or repository.
The replaced histories are not continuous and seem random.
This is indeed a puzzling situation. To get to the bottom of it, you might want to consider the following steps:

Review Access Permissions: Ensure that there haven't been any unauthorized accesses to your repositories. Check if any contributors or collaborators have been added to your repositories without your knowledge.

Email Notifications: Review the email notifications sent by GitHub or your Git service. They might contain information about changes made to your repositories.

Change Account Credentials: As a precaution, consider changing the password and security settings for both accounts.

GitHub Support: Reach out to GitHub or the Git service's support team with this issue. They might be able to provide further insights into what's happening.

Security Audit: Run a security audit on your repositories to check for any suspicious activity.

Check for Scripts or Automation: Ensure that no scripts or automation tools are causing these changes.
For more details visit **[my website](https://srd-status-check.co.za/)**
